pkg/cmd/build/errors.go

```diff
@@ -86,7 +91,7 @@ func isTransientError(err error) bool {
 
 // IsLoggedIntoRegistryButDontHavePermissions returns true when the error is because the user is logged into the registry but doesn't have permissions to push the image
 func isLoggedIntoRegistryButDontHavePermissions(err error) bool {
-	return strings.Contains(err.Error(), "insufficient_scope: authorization failed")
+	return strings.Contains(err.Error(), "insufficient_scope: authorization failed") && !isPullAccessDenied(err)
```

Comment:
the full errror for Pull Access Denied includes "insufficient_scope: authorization failed" so I made explicit that `isLoggedIntoRegistryButDontHavePermissions` should ignore that part.

We could move above the check for `isPullAccessDenied` but it seems better excluding IMHO.